### ISSUE TITLE
IRT and Random Selection

### ISSUE THREAD

**Issue by Re-dot-art** [bigdata-ustc/EduCDM#42]
The provided item response theory dataset was run twice, taking the top 10% of the most difficult questions b, and the repetition rate of the results obtained twice was less than 5%. This proves that IRT approximates random selection. Why is this?

**Comment by Re-dot-art** [comment]
Is it because there are too few epoches of training? 3 epoches of training in the code

**Comment by Re-dot-art** [comment]
> Hi, there, please refer to the script you are using. I am not sure where your problem comes from.

https://github.com/bigdata-ustc/EduCDM/blob/main/examples/IRT/GD/IRT.py , well, I ran this code and output self. a, self. b, and self. c. It was found that the difficulty of the project b was not consistent between the two runs for three rounds. That is, for a project, the results of the two runs are inconsistent, and there is a significant difference.

**Comment by tswsxk** [comment]
We set the epoch as 2 just for a demo case. In real scenario, the number of epochs is a hyper-parameter which should be tuned based on the result on the validation dataset.

**Comment by Re-dot-art** [comment]
> We set the epoch as 2 just for a demo case. In real scenario, the number of epochs is a hyper-parameter which should be tuned based on the result on the validation dataset.

Thank you for your reply. I tried to increase the number of epochs to 60, but the number of repeatedly selected samples has increased to 890/1700, but it is still only about 50%.

**Comment by tswsxk** [comment]
I am not very sure I understand your problem. You are checking the parameters generated with GDIRT, and try to find the top 10% of the most difficult questions. To achieve that, you select the top 10% at each training epoch and calculate the overlap?

**Comment by Re-dot-art** [comment]
> I am not very sure I understand your problem. You are checking the parameters generated with GDIRT, and try to find the top 10% of the most difficult questions. To achieve that, you select the top 10% at each training epoch and calculate the overlap?

Yes, I think I already know what the problem is. The data volume should be too large, requiring multiple epochs to fit. Thank you for your reply!
